src/controllers/project.js

```diff
@@ -46,6 +48,98 @@ const supportProject = async (req, res, next) => {
   return next();
 };
 
+const createReview = async (req, res, next) => {
+  const { id } = req.params;
+  const { userID, rating, content } = req.body;
+
+  try {
+    const project = await Project.findOne({ _id: id });
+
+    if (!project) return res.status(404).json({ message: 'Project not found' });
+
+    const review = {
+      user: userID,
+      rating,
+      content,
+    };
+
+    project.reviews.push(review);
+
+    await project.save();
+
+    res.json({
+      message: 'Review created successfully',
+      reviews: project.reviews,
+    });
+  } catch (error) {
+    res.status(422).json({ message: 'Unable to create review' });
+  }
+  return next();
+};
+
+const isEmpty = (field) => !!field;
+
+const updateReview = async (req, res, next) => {
+  const { id, reviewId } = req.params;
+  const { rating, content } = req.body;
+
+  try {
+    const project = await Project.findById(id);
+
+    if (!project) return res.status(404).json({ message: 'Project not found' });
+
+    const review = project.reviews.find(
+      // eslint-disable-next-line no-underscore-dangle
+      (single) => ObjectId(single._id).toString() === reviewId
+    );
+
+    if (!review) return res.status(404).json({ message: 'Review not found' });
+
+    review.rating = isEmpty(rating) ? rating : review.rating;
+    review.content = isEmpty(content) ? content : review.content;
```

Comment:
here the assignation would be reversed when that one is changed to one negation in both lines

Reply:
if not empty take the new rating/content from req.body else keep the current one

Reply:
this is tested and its working